packages/databricks-vscode/src/configuration/loaders/OverrideableConfigLoaderWriter.ts

```diff
@@ -0,0 +1,73 @@
+import {StateStorage} from "../../vscode-objs/StateStorage";
+import {OverrideableConfig} from "../types";
+import {CachedValue} from "../../locking/CachedValue";
+import {Disposable} from "vscode";
+import {Mutex} from "../../locking";
+
+export class OverrideableConfigLoaderWriter implements Disposable {
```

Comment:
`loaders/OverrideableConfigLoaderWriter.ts` - nitpick, but should the "writer" be in the loader directory? Can we move things from here to one level higher and just name things as `*Model` (or without "model", pretty much the same as we had before)?

Reply:
I agree. I refactored this. Also refactored a bunch of other code to be more streamlined. Ig we can make the next PR a bit more streamlined as well! Thanks for the patience here :)